plugins/CMakeLists.txt

```
@@ -2,9 +2,44 @@
 include_directories(SYSTEM
   ${ODB_INCLUDE_DIRS})
 
+if(WITH_PLUGIN AND WITHOUT_PLUGIN)
+  message(FATAL_ERROR "The WITH_PLUGIN and WITHOUT_PLUGIN options cannot be used together.")
+endif()
+
 # Add all subdirectories to the build
 file(GLOB plugins RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "${CMAKE_CURRENT_SOURCE_DIR}/*")
 
+# Remove subdirectories according to the provided WITH_PLUGIN or WITHOUT_PLUGIN variables
+if(WITH_PLUGIN)
+  if(NOT ("metrics" IN_LIST WITH_PLUGIN)
+      OR NOT ("search" IN_LIST WITH_PLUGIN))
+    message(WARNING "The search and metrics plugins cannot be left out, "
+            "they will be compiled anyway.")
```

Emphasise they are options:

```suggestion
    message(WARNING "The 'search' and 'metrics' plugins cannot be left out, "
            "they will be compiled anyway.")
```